machine/memmgnt/memmgnt.go

```diff
@@ -0,0 +1,122 @@
+package memmgnt
+
+import (
+	"errors"
+	"syscall"
+	"unsafe"
+
+	"github.com/bobuhiro11/gokvm/kvm"
+)
+
+var (
+	errNoSlotsAvail = errors.New("maximal numbers of slots exhausted")
+	errSlotNotFound = errors.New("unable to find MemorySlot")
+)
+
+const (
+	// Poison is an instruction that should force a vmexit.
+	// it fills memory to make catching guest errors easier.
+	// vmcall, nop is this pattern
+	// Poison = []byte{0x0f, 0x0b, } //0x01, 0xC1, 0x90}
+	// Disassembly:
+	// 0:  b8 be ba fe ca          mov    eax,0xcafebabe
+	// 5:  90                      nop
+	// 6:  0f 0b                   ud2
+	Poison = "\xB8\xBE\xBA\xFE\xCA\x90\x0F\x0B"
+
+	highMemBase = 0x100000
+)
+
+type RegionType uint8
+
+const (
+	RAM RegionType = 0 + iota
+	ROM
+	IO
+	PCI
+	VGA
+)
+
+type MemMgnt struct {
+	Slots
+	MaxSlots uint32
+}
+
+type MemorySlot struct {
+	Addr          uint64
+	Size          int
+	Slot          uint8
+	Flags         uint32
+	OldFlags      uint32
+	DirtyBMap     uint32
+	DirtyBMapSize uint32
+	AsID          uint32
+	PhysAddr      uint64
+	Buf           []byte
+}
+
+type Slots []*MemorySlot
```

Comment:
I don't think this type declaration is necessary because it doesn't make much sense.
(Will it be necessary depending on future implementations?)

Reply:
Not 100% sure yet. Probably some operations in regards of sorting, find particular slots etc may require it.

Reply:
I can also implement this operations on the memory struct. Hence you are right by requesting to remove it